wordpress-mobile/WordPress-Android
# [Gutenberg] - Filter media types when sharing files to the app

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/6443

**Related PRs:**
- https://github.com/WordPress/gutenberg/pull/56791

This PR fixes an issue when appending unsupported media types when using the Sharing functionality on Android devices.

The only way to be able to "Share" unsupported media types is when multi-selecting files that the app does support like images or videos.

For more information please check Gutenberg's PR description.

-----

## To Test:

- **Precondition**: Have different kinds of files on your device: `PDF`, `jpg`,  and `mp4`.
- **NOTE:** Since this is not a release build, the editor takes a bit of time to load when adding media.
- Install the app from this PR
- Open your Files app
- Multiselect an image and a PDF file 
- Tap on the Share button
- Select the Jetpack app
- **Expect** the editor to be loaded with just the supported media types (images, and videos)

Test Case 1|Test Case 2
-|-
<video src="https://github.com/WordPress/gutenberg/assets/4885740/dd8dd610-5371-4da0-acf2-a36d1914c97e" />|<video src="https://github.com/wordpress-mobile/WordPress-Android/assets/4885740/19574a54-0db9-41bf-ab45-e30e8cb80278" />

-----

## Regression Notes

1. Potential unintended areas of impact

It should only affect the editor.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual tests.

3. What automated tests I added (or what prevented me from doing so)

    - TODO

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
